src/universal_transfer_operator/data_providers/constants.py

```diff
@@ -0,0 +1,71 @@
+from enum import Enum
+
+
+class SupportedDataProvider(Enum):
+    """
+    List of all the supported DataProviders.
+    """
+
+    LOCAL = "local"
+    S3 = "s3"
+    FTP = "ftp"
+    HDFS = "hdfs"
+    SFTP = "sftp"
+    GCS = "gcs"
+    MYSQL = "mysql"
+    POSTGRES = "postgres"
+    BIGQUERY = "bigquery"
+    REDSHIFT = "redshift"
+
+
+class Database(Enum):
+    """
+    List of all the supported Databases DataProviders
+    """
+
+    POSTGRES = "postgres"
+    POSTGRESQL = "postgres"
+    SQLITE = "sqlite"
+    BIGQUERY = "bigquery"
+    SNOWFLAKE = "snowflake"
+
+
+class FileSystem(Enum):
+    """
+    List of all the supported FileSystem DataProviders
+    """
+
+    LOCAL = "local"
+    GCS = "gcs"  # Google Cloud Storage
+    S3 = "s3"  # Amazon S3
```

Comment:
There are times when a user is simply downloading a file from an HTTP/HTTPSs endpoint - as opposed to accessing an API with credentials and paginating.  I believe we could benefit by having both HTTP & HTTPs options here.

Reply:
But when a user downloads from an HTTP/HTTPS endpoint, isn't it a `GET` REST call? Or am I understanding it wrong?